Code/Framework/AzToolsFramework/AzToolsFramework/Entity/PrefabEditorEntityOwnershipService.cpp

```diff
@@ -104,39 +109,49 @@ namespace AzToolsFramework
     void PrefabEditorEntityOwnershipService::AddEntity(AZ::Entity* entity)
     {
         AZ_Assert(IsInitialized(), "Tried to add an entity without initializing the Entity Ownership Service");
-        ScopedUndoBatch undoBatch("Undo adding entity");
+
+        // Setup undo node.
+        ScopedUndoBatch undoBatch("Add entity");
+
+        // Determine which prefab instance should own this entity.
+        Prefab::InstanceOptionalReference newOwningInstance = m_prefabFocusInterface->GetFocusedPrefabInstance(m_entityContextId);
```

Comment:
repeated: nullopt check would be good here before trying to access the reference.

Reply:
Fixed!